Mathlib/LinearAlgebra/Matrix/SesquilinearForm.lean

```diff
@@ -169,37 +174,37 @@ def LinearMap.toMatrixₛₗ₂' : ((n → R₁) →ₛₗ[σ₁] (m → R₂) 
     right_inv := Matrix.toMatrix₂Aux_toLinearMap₂'Aux }
 #align linear_map.to_matrixₛₗ₂' LinearMap.toMatrixₛₗ₂'
 
-/-- The linear equivalence between bilinear forms and `n × m` matrices -/
-def LinearMap.toMatrix₂' : ((n → R) →ₗ[R] (m → R) →ₗ[R] R) ≃ₗ[R] Matrix n m R :=
+/-- The linear equivalence between bilinear maps and `n × m` matrices -/
+def LinearMap.toMatrix₂' : ((n → R) →ₗ[R] (m → R) →ₗ[R] N₂) ≃ₗ[R] Matrix n m N₂ :=
   LinearMap.toMatrixₛₗ₂'
 #align linear_map.to_matrix₂' LinearMap.toMatrix₂'
 
 variable (σ₁ σ₂)
 
-/-- The linear equivalence between `n × n` matrices and sesquilinear forms on `n → R` -/
-def Matrix.toLinearMapₛₗ₂' : Matrix n m R ≃ₗ[R] (n → R₁) →ₛₗ[σ₁] (m → R₂) →ₛₗ[σ₂] R :=
+/-- The linear equivalence between `n × n` matrices and sesquilinear maps on `n → R` -/
+def Matrix.toLinearMapₛₗ₂' : Matrix n m N₂ ≃ₗ[R] (n → R₁) →ₛₗ[σ₁] (m → R₂) →ₛₗ[σ₂] N₂ :=
   LinearMap.toMatrixₛₗ₂'.symm
 #align matrix.to_linear_mapₛₗ₂' Matrix.toLinearMapₛₗ₂'
 
-/-- The linear equivalence between `n × n` matrices and bilinear forms on `n → R` -/
-def Matrix.toLinearMap₂' : Matrix n m R ≃ₗ[R] (n → R) →ₗ[R] (m → R) →ₗ[R] R :=
+/-- The linear equivalence between `n × n` matrices and bilinear maps on `n → R` -/
+def Matrix.toLinearMap₂' : Matrix n m N₂ ≃ₗ[R] (n → R) →ₗ[R] (m → R) →ₗ[R] N₂ :=
   LinearMap.toMatrix₂'.symm
 #align matrix.to_linear_map₂' Matrix.toLinearMap₂'
 
-theorem Matrix.toLinearMapₛₗ₂'_aux_eq (M : Matrix n m R) :
+theorem Matrix.toLinearMapₛₗ₂'_aux_eq (M : Matrix n m N₂) :
     Matrix.toLinearMap₂'Aux σ₁ σ₂ M = Matrix.toLinearMapₛₗ₂' σ₁ σ₂ M :=
   rfl
 #align matrix.to_linear_mapₛₗ₂'_aux_eq Matrix.toLinearMapₛₗ₂'_aux_eq
 
-theorem Matrix.toLinearMapₛₗ₂'_apply (M : Matrix n m R) (x : n → R₁) (y : m → R₂) :
+theorem Matrix.toLinearMapₛₗ₂'_apply (M : Matrix n m N₂) (x : n → R₁) (y : m → R₂) :
     -- porting note: we don't seem to have `∑ i j` as valid notation yet
-    Matrix.toLinearMapₛₗ₂' σ₁ σ₂ M x y = ∑ i, ∑ j, σ₁ (x i) * M i j * σ₂ (y j) :=
+    Matrix.toLinearMapₛₗ₂' σ₁ σ₂ M x y = ∑ i, ∑ j, σ₁ (x i) • σ₂ (y j) • M i j :=
```

Comment:
Should this be an `op` smul so that it actually generalizes what was there before? Maybe with appropriate `IsCentralScalar` assumptions or something.

Reply:
Would this be using `GroupTheory/GroupAction/Opposite`?